### DISCUSSION TITLE
EC11-Rotary-Encoder  bad working

### DISCUSSION THREAD

**Discussion by axel-GB** [arendst/Tasmota#21499] [General]
Hi, i have bought [https://pl.aliexpress.com/item/1005004109624610.html?spm=a2g0o.order_list.order_list_main.17.18f31c245OgV5N&gatewayAdapt=glo2pol](url)

It doesn't want to work well, i.e. PWM dimming. Instead of dimming, it turns the LED on and off randomly.
Only after a long time I discovered that EC11 works fine as long as SetOption68 is OFF.
When So68 is 1, dimming problems begin.

I have more PWM's than 1 and So68 must be 1.
Please Help.

**Comment by sfromis** [comment]
Well,, if you have `SetOption68 on` you do just not get any dimming (meaning the Dimmer command), and the rotary is not relevant, unless you create rules or other automation to connect the dots.

Tasmota only supports **one** light. What you get with `SetOption68` is a special mode where it is no longer lights, but channels, handled in a different way than lights.

**Reply by axel-GB** [reply]
I Have tested new version of Tasmota 14.3.0 and have an idea to test my Encoder.

It works bad with only 1 PWM, so68 0, 
factory settings only  PWM1  , Rotary_A_n 1 , Rotary B_n 1, button 1

Push button works, and it can only dimmer - in booth ways, right or left allways DIMMINIG - .
14:28:53.055 RSL: RESULT = {"POWER":"ON","Dimmer":12}
14:29:04.654 RSL: RESULT = {"POWER":"OFF","Dimmer":0}         - right 
14:29:08.162 RSL: RESULT = {"POWER":"OFF","Dimmer":0}         - left
14:29:11.137 RSL: RESULT = {"POWER":"OFF","Dimmer":0}          -right 
why is other direction of encoder not working for dimmer + .

**Reply by sfromis** [reply]
That rotary does not look like the no pull-up option is the right one, try with `Rotary_A` instead of `Rotary_A_n`, and likewise for B. And check your value for `SetOption43` to see available range for the Dimmer mode.
